mapproxy/cache/path.py

```diff
@@ -223,4 +228,64 @@ def tile_location_arcgiscache(tile, cache_dir, file_ext, create_dir=False):
     return tile.location
 
 def level_location_arcgiscache(z, cache_dir):
-    return level_location('L%02d' % z, cache_dir=cache_dir)
\ No newline at end of file
+    return level_location('L%02d' % z, cache_dir=cache_dir)
+
+def tile_location_gwc(tile, cache_dir, file_ext, grid_name, create_dir=False):
+    """
+    Return the location of the `tile`. Caches the result as ``location``
+    property of the `tile`.
+
+    :param tile: the tile object
+    :param grid_name: The grid name that act as a prefix in level directory names
+    :param create_dir: if True, create all necessary directories
+    :return: the full filename of the tile
+
+    >>> from mapproxy.cache.tile import Tile
+    >>> tile_location_gwc(Tile((0, 0, 0)), '/tmp/cache', 'png', 'EPSG:2163').replace('\\\\', '/')
+    '/tmp/cache/EPSG_2163_00/0_0/00_00.png'
+    """
+
+    from math import ceil, log10
+
+    def gwczeropadder(number, order):
```

Comment:
Please do not port code directly form geowebcache, since the license is incompatible. 
Anyhow, you should have ported the comment as well: "Silly way to pad numbers with leading zeros, since I don't know a fast way of doing this in Java."

Not sure what this does, by you can pad in Python with `'{:012d}'.format(123)` 😄